src/main/java/core/basesyntax/strategy/PurchaseHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class PurchaseHandler implements OperationHandler {
+    private static final String PURCHASE = "p";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(PURCHASE)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
```

Comment:
The variable 'oldQuantity' is reset to 0 for each 'listOfFruit' iteration, which seems incorrect. It should maintain the state of the total available quantity for each fruit type. Also, the logic for updating the quantity in line 18 is incorrect; it should subtract the purchase quantity from the available quantity.